### ISSUE TITLE
missing-parameter false positive for a table with both __call and __index metamethods

### ISSUE THREAD

**Issue by ligurio** [EmmyLuaLs/emmylua-analyzer-rust#1251]
Imagine a file `metatable_call.lua` with the following Lua chunk:

```lua
local function __call(self, opts)
    return self
end

---@param key string
---@return any
local function __index(_, key)
    return nil
end

local cfg = setmetatable({}, {__index = __index, __call = __call})
cfg({})
```

and the following config `.emmyrc.json` for the emmylua_check: `{ "runtime": { "version": "LuaJIT" } }`.

Run the command: `emmylua_check --config .emmyrc.json metatable_call.lua`
The command reports:

```
---  [1 warning, 2 hints]
hint: key is never used, if this is intentional, prefix it with an underscore: _key [unused]
  -->: :7:27

  6 │ ---@return any
  7 │ local function __index(_, key)
  8 │     return nil

hint: opts is never used, if this is intentional, prefix it with an underscore: _opts [unused]
  -->: :1:29

  1 │ local function __call(self, opts)
  2 │     return self

warning: expected 2 parameters but found 1. missing parameter: key [missing-parameter]
  -->: :12:7

  11 │ local cfg = setmetatable({}, {__index = __index, __call = __call})
  12 │ cfg({})



Summary
  1 warning
  2 hints

Check completed with warnings
Check finished
```

It is expected no diagnostics.

It seems calling a table that has both `__call` and `__index` metamethods is resolved to `__index(_, key)` instead of `__call(self, opts)`, hence the spurious missing parameter: `key`.

Version: emmylua_check 0.24.0
